runtime/vault-filesystem/src/main/java/io/aklivity/zilla/runtime/vault/filesystem/internal/FileSystemVaultHandler.java

```diff
@@ -56,6 +58,15 @@ public FileSystemVaultHandler(
         lookupTrust = supplyLookupTrustedCertificateEntry(resolvePath, options.trust);
         lookupSigner = supplyLookupTrustedCertificateEntry(resolvePath, options.signers);
         lookupKeys = supplyLookupPrivateKeyEntries(resolvePath, options.keys);
+        resources = new LinkedList<>();
+        if (options.keys != null && options.keys.store != null)
+        {
+            resources.add(options.keys.store);
+        }
+        if (options.trust != null && options.trust.store != null)
+        {
+            resources.add(options.trust.store);
+        }
```

Comment:
We can introduce the concept of `resources` closer to the parsed config to reduce complexity.

The overall design has 1 engine, with N engine workers (1 per core).
Similarly, we have 1 engine config, with N engine registries for N workers.

The engine config consists of a set of namespace configs, which are parsed from `zilla.yaml`, potentially with multiple namespaces if `---` separator is used for a multi-document yaml file.

Then it would seem natural for each namespace config to understand the resources configured by any of its components, such as local files retrieved by the filesystem vault for example.

This might most naturally by conveyed to the namespace for the parsed vault config by the filesystem vault itself, as only the more specific component knows what resources it needs.

Each per worker engine registry should not need to independently watch either the zilla.yaml config nor the individual resources. We need change detection of both zilla.yaml and resources to be done independent of any specific worker, as it is now for zilla.yaml, executed as a task on a scheduled executor service, based on resources conveyed by shared engine config, rather than per engine worker handlers.

Also, let's aim to have a single jdk filesystem watch service in use, rather than a separate one for config and a different one for resources.

Effectively, we now need an engine config watcher that watches all relevant aspects of the combined engine config, potentially with multiple namespace configs and resource configs.